ksql-engine/src/test/resources/query-validation-tests/struct-udfs.json

```diff
@@ -0,0 +1,33 @@
+{
+  "comments": [
+    "Tests covering the use of the struct creation/access in UDFs (exclusively test methods)"
+  ],
+  "tests": [
+    {
+      "name": "Create a struct from a string",
+      "statements": [
+        "CREATE STREAM test (value STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT TOSTRUCT(value) AS value FROM test;"
```

Comment:
I wonder if `TOSTRUCT` wouldn't actually be a useful production UDF!  ;)

```SQL
SELECT AS_STRUCT(col0, "price", col3, "vat") FROM blah
```

(Exact naming TBD...)

Reply:
See discussion below in my main comment - being able to determine the strict output type at runtime is non-trivial.